lib/crypto/c_src/bn.c

```diff
@@ -55,67 +79,108 @@ ERL_NIF_TERM bin_from_bn(ErlNifEnv* env, const BIGNUM *bn)
     ERL_NIF_TERM term;
 
     /* Copy the bignum into an erlang binary. */
-    bn_len = BN_num_bytes(bn);
-    bin_ptr = enif_make_new_binary(env, bn_len, &term);
-    BN_bn2bin(bn, bin_ptr);
+    if ((bn_len = BN_num_bytes(bn)) < 0)
```

Comment:
How could this possible be negative? The man pages does not say that it can return `< 0` and the source code in OpenSSL suggests that it can not return a negative value.

Reply:
In order to remove that check and still cast to `size_t`, I would need to check all versions of OpenSSL that Erlang supports and be confident that they never introduce a negative value as an error condition in the future.  The man page does not say it cannot return `< 0`.  The man page doesn't say anything about possible error conditions.